server/controllers/request.js

```diff
@@ -0,0 +1,79 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler")
+
+// @route POST /requests/create
+// @desc Create new request
+// @access Public
+exports.createRequest = asyncHandler(async (req, res, next) => {
+  const request = await Request.create(req.body);
+
+  if (request) {
+    res.status(200).json({
+      success: {
+        request: {
+          ...request
+        }
+      }
+    })
+  } else {
+    res.status(400);
+    throw new Error("Invalid request data")
+  }
+});
+
+// @route PUT /request/edit
+// @desc edit request
+// @access Public
+exports.editRequest = asyncHandler(async (req, res, next) => {
+  const request = await Request.findOne({ _id: req.request._id })
+  
+  if (!request) {
+    res.status(404);
+    throw new Error("Request doesn't exist");
+  }
+  request.set(req.body);
+  const updatedRequest = await request.save();
+  res.status(200).json({
+    success: {
+      request: updatedRequest
+    },
+  });
+});
+
+// @route GET /requests
+// @desc get requests of logged in user
+// @access Private
+exports.getUserRequests = asyncHandler(async (req, res, next) => {
+  const requests = await Request.where('requester', req.userId);
+
+  if (!requests) {
+    res.status(404);
+    throw new Error("User has no requests or is not logged in")
+  }
```

Comment:
There are a couple of issues with this:
- [`model.where/find`](https://mongoosejs.com/docs/api/model.html#model_Model.where) will return an array when it suceeds (sometimes empty, sometimes not). And if it fails (i.e. the query rejects), the `await` keyword will throw an error as mentioned before. So in no case will `!requests` evaluate to true (remember that empty arrays are [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) too)!
- If the list of requests for the user is empty, we shouldn't respond with a `404`. `404` implies we couldn't locate the resource, which is not true in this case - we located it all fine but it's just empty. So it would be more appropriate to just send the empty collection with a success status code 👍🏼 (you can find more on this [here](https://stackoverflow.com/a/13367198/15220729))

Reply:
Makes sense